doc/rst/language/evolution.rst

```diff
@@ -3,18 +3,225 @@
+  simply update the specific access, but use it as a cue to look
+  through the code for other references to that variable that will
+  also need updating.
+
+* When possible, try rewriting your updated code to use an
+  index-neutral style of programming.  For example, given code like
+  this:
+
+  .. code-block:: chapel
+
+      var t: 2*int = ...;
+
+      var x = t(1),
+          y = t(2);
+
+      for i in 1..2 do
+        writeln("t(", i, ") = ", t(i));
+
+  It would be reasonable to rewrite it like this:
+
+  .. code-block:: chapel
+
+      var t: 2*int = ...;
+
+      var x = t(0),
+          y = t(1);
+
+      for i in 0..1 do
+        writeln("t(", i, ") = ", t(i));
+
+  But arguably preferable to update it like this:
+
+  .. code-block:: chapel
+
+      var t: 2*int = ...;
+
+      var (x, y) = t;
+
+      for i in t.indices do
+        writeln("t(", i, ") = ", t(i));
+
+  If you have a pattern that you're trying to write in an
+  index-neutral style, but can't, don't hesitate to ask for tips.
+        
+
+* Some common pitfalls to check for in your code include:
+
+  - Search for queries on the dimensions of rectangular domains and
+    arrays.  For example, ``myDomain.dim(1)``, ``myDomain.low(1)``,
+    ``myDomain.high(1)``, or ``myDomain.stride(1)`` will need to be
+    updated to reflect that array dimensions now count from 0 rather
+    than 1.  These will result in out-of-bounds errors in cases where
+    you query all dimensions of an array, making them easy to find;
+    but it can be worthwhile to grep your code for such patterns to
+    make sure you don't miss any.
+
+  - Also search for instances of ``find()`` or ``rfind()`` that are
+    relying on comparisons to zero/nonzero values, and update them to
+    compare against -1.  For example, patterns like ``if
+    mystring.find('z')`` need to be updated to ``if mystring.find('z')
+    != -1``.
+
+  - Search for instances of ``split()``.  A common idiom is to write
+    ``var substrs = mystring.split(5);`` and then to index into the
+    result using ``substrs[1]``, ``substrs[2]``, etc.  Since this is
+    an instance of capturing an iterator expression, you'll either
+    need to subtract one from the indices, or else declare `substrs`
+    to have a specific type, like ``var substrs: [1..5] string =
+    mystring.split(5);``
+
+  - Search for varargs functions and make sure they are updated to use
+    0-based indexing or index-neutral features.
+
+  - Search for any calls to ``Reflection.getField*()`` and update
+    those the cases that use integer indices to reflect 0-based
+    numbering.
+
+  - Look for any calls on lists that use explicit offsets, as these
+    will likely need updates.  For example ``mylist.pop(1);`` will
+    need to become ``mylist.pop(0);``
+
+Need Help?
+**********
+
+If you are able to share your code with us and would like help
+updating it to Chapel 1.22, please don't hesitate to ask for help.
```

Comment:
How do we recommend people reaching out? Mailing list?

Reply:
Let's point to the chapel users page here too if that sounds OK.

Reply:
Sounds good